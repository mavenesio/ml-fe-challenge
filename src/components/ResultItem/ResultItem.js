import React from 'react';
import Styled from 'styled-components';
import freeShippingIcon from '../../assets/ic_shipping.png';
import currencyEnum from '../../constants/currency';

const ResultContainer = Styled.div`
  width:70%;
  align-self:center;
  border-radius:4px;
  border-bottom: 1px solid gray;
  background-color:white;
  display:flex;
  flex-direction:row;
  justify-content:flex-start;
`;
const ResultImageContainer = Styled.div`
  border-radius: 4px;
  padding:16px;

`;
const ResultBody = Styled.div`
  padding-bottom: 1rem;
  display:flex;
  flex-direction:row;
  justify-content:space-between;
`;
const ResultPrice = Styled.div`
  align-self:flex-start;
  font-size:24px;
  margin: 1rem 0rem;
`;
const ResultTitle = Styled.div`
  align-self:flex-start;
  font-size:18px;
`;
const ResultFirstColumn = Styled.div`
  display:flex;
  flex-direction:column;
  justify-content:flex-start;
  margin-right:1rem;
  width:350px;
`;
const ResultSecondColumn = Styled.div`
  margin-top: 3rem;
  padding:1rem;
  align-self:start;
`;
const ResultItem = ({id, thumbnail, currency, price, freeShipping, title, stateName}) =>  {
  return (
    <ResultContainer key={id}>
    <ResultImageContainer>
      <img src={thumbnail} width='180px' height='180px' alt=''/>
    </ResultImageContainer>
    <ResultBody>
      <ResultFirstColumn>
        <ResultPrice>
            {`${currencyEnum[currency]} ${price.toFixed().replace(/\B(?<!\.\d*)(?=(\d{3})+(?!\d))/g, ".")} `}
            {
              freeShipping &&
              <img src={freeShippingIcon} width='20px' height='20px' alt=''/>
            }
        </ResultPrice>
        <ResultTitle>
            {title}
        </ResultTitle>
      </ResultFirstColumn>
      <ResultSecondColumn>
            {stateName}
      </ResultSecondColumn>
    </ResultBody>
  </ResultContainer>
  );
}

export default ResultItem;
