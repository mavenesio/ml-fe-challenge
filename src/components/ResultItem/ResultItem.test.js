import React from 'react';
import ResultItem from './ResultItem';

describe('ResultItem', () => {
  it('should render without props', () => {   
    expect(<ResultItem />).toMatchSnapshot();
  });

  it('should render with props', () => {
    expect(<ResultItem id='MLA846526085' 
                       thumbnail='http://mla-s2-p.mlstatic.com/662389-MLA31003000984_062019-I.jpg'
                       currency='$'
                       price={139989.99}
                       freeShipping={true}
                       title='Apple iPhone 8 Plus 64 Gb Gris Espacial 3 Gb Ram'
                       stateName='Capital Federal'  />).toMatchSnapshot();
  });
});




