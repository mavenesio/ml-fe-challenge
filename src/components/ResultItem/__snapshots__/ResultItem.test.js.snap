// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResultItem should render with props 1`] = `
<ResultItem
  currency="$"
  freeShipping={true}
  id="MLA846526085"
  price={139989.99}
  stateName="Capital Federal"
  thumbnail="http://mla-s2-p.mlstatic.com/662389-MLA31003000984_062019-I.jpg"
  title="Apple iPhone 8 Plus 64 Gb Gris Espacial 3 Gb Ram"
/>
`;

exports[`ResultItem should render without props 1`] = `<ResultItem />`;
